plugins/browser-plugin-youtube-tracking/src/index.ts

```diff
@@ -0,0 +1,231 @@
+  tag.src = 'https://www.youtube.com/iframe_api';
+  const firstScriptTag = document.getElementsByTagName('script')[0];
+  firstScriptTag.parentNode!.insertBefore(tag, firstScriptTag);
+}
+
+export function enableYouTubeTracking(args: { id: string; options?: MediaTrackingOptions }) {
+  const conf: TrackingOptions = trackingOptionsParser(args.id, args.options);
+  const el: HTMLIFrameElement = document.getElementById(args.id) as HTMLIFrameElement;
+  if (!el) {
+    LOG.error('Cannot find YouTube IFrame element');
+    return;
+  }
+  conf.urlParameters = new UrlParameters(el.src);
+  el.src = conf.urlParameters.getUrl();
+  enableYouTubeIframeAPI();
+
+  // Function to run when the API is loaded
+  window.onYouTubeIframeAPIReady = () => {
+    iframeApiHasLoaded = true;
+    trackingQueue.forEach((conf) => {
+      addListeners(conf);
+    });
+  };
+
+  // If the API hasn't loaded, it will be added to the queue
+
+  if (iframeApiHasLoaded) {
+    addListeners(conf);
+  } else {
+    trackingQueue.push(conf);
+  }
+}
+
+function addListeners(conf: TrackingOptions) {
+  const builtInEvents: Record<string, Function> = {
+    [YTPlayerEvent.ONREADY]: () => youtubeEvent(tracked_players[conf.mediaId].player, YTEvent.READY, conf),
+    [YTPlayerEvent.ONSTATECHANGE]: (e: YT.OnStateChangeEvent) => {
+      if (conf.captureEvents.indexOf(YTStateEvent[e.data.toString()]) !== -1) {
+        youtubeEvent(tracked_players[conf.mediaId].player, YTStateEvent[e.data], conf);
+      }
+    },
+    [YTPlayerEvent.ONPLAYBACKQUALITYCHANGE]: () =>
+      youtubeEvent(tracked_players[conf.mediaId].player, YTEvent.PLAYBACKQUALITYCHANGE, conf),
+    [YTPlayerEvent.ONAPICHANGE]: () => youtubeEvent(tracked_players[conf.mediaId].player, YTEvent.APICHANGE, conf),
+    [YTPlayerEvent.ONERROR]: (e: YT.OnErrorEvent) =>
+      youtubeEvent(tracked_players[conf.mediaId].player, YTEvent.ERROR, conf, { error: YTError[e.data] }),
+    [YTPlayerEvent.ONPLAYBACKRATECHANGE]: () =>
+      youtubeEvent(tracked_players[conf.mediaId].player, YTEvent.PLAYBACKRATECHANGE, conf),
+  };
+
+  let playerEvents: Record<string, Function> = {};
+  for (let event of conf.youtubeEvents) {
+    playerEvents[event] = builtInEvents[event];
+  }
+
+  tracked_players[conf.mediaId] = {
+    player: new YT.Player(conf.mediaId, { events: { ...playerEvents } }),
+    conf: conf,
+    seekTracking: {
+      prevTime: 0,
+      enabled: false,
+    },
+    volumeTracking: {
+      prevVolume: 0,
+      enabled: false,
+    },
+  };
+}
+
+function youtubeEvent(player: YT.Player, eventName: string, conf: TrackingOptions, eventData?: EventData) {
+  let playerInstance = tracked_players[conf.mediaId];
+  if (!playerInstance.seekTracking.enabled && conf.captureEvents.indexOf('seek') !== 1)
+    enableSeekTracking(player, conf, eventData);
+  if (!playerInstance.volumeTracking.enabled && conf.captureEvents.indexOf('volume') !== 1)
+    enableVolumeTracking(player, conf, eventData);
```

Comment:
Lets get some brackets around these I think? I'm sort of surprised prettier didn't do this automatically?